### PR TITLE
[GPU]Fix layout count comparison by using actual byte value in realloc_if_needed

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Fix layout count comparison by using actual byte value

### Tickets:
 - 144293

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 1]
```diff
@@ -556,7 +556,7 @@ event::ptr primitive_inst::realloc_if_needed() {
         // For nodes that can be optimized, variable memory is used as output memory
         // so there is no need for output memory reallocation
         if (can_be_optimized()) {
-            _max_output_layout_count = variable.get_actual_mem_size() / (dt_size / 8);
+            _max_output_layout_count = variable.get_actual_mem_size();
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/primitive_inst.cpp:559]: This change won't be entirely correct as in some cases _max_output_layout_count  still operates with byte size. That part for instance
```cpp
        if (can_be_optimized()) {
            _max_output_layout_count = variable.get_actual_mem_size();
```

So please update all relevant parts of the code

**Review by vladimir-paramuzov — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp (hunk 1 above)
Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp:559]: @vladimir-paramuzov My intension is using byte size for _max_output_layout_count. variable.get_actual_mem_size() returns _bytes_count from cldnn::memory structure. Is there any code that _max_output_layout_count is not using byte size? Or I need to change name of _max_output_layout_count to be clearer?

Comment (yeonbok) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp:559]: @wilson-seok  I dont understand why we should use byte size for _max_output_layout_count. As I see all the codes are using "count" as max_output_layout_count.

Comment (wilson-seok) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp]: @vladimir-paramuzov  @yeonbok Understood! I updated those codes that _max_output_layout_count has element size including padding, Thanks!

**Review by vladimir-paramuzov — APPROVED** [review]
LGTM. Please ensure that it doesn't impact the performance as get_buffer_size() call is more expensive

**Pull request merged**